public/app/features/trails/DataTrailsHistory.tsx

```diff
@@ -18,6 +18,7 @@ import { VAR_FILTERS } from './shared';
 import { getTrailFor } from './utils';
 
 export interface DataTrailsHistoryState extends SceneObjectState {
+  currentStep: number;
```

Comment:
Making the current step part of the history state, and updating it at each transition helped trigger re-rendering the history component.